api/src/main/java/org/openmrs/module/idgen/LocationBasedPrefixProvider.java

```diff
@@ -0,0 +1,63 @@
+package org.openmrs.module.idgen;
+
+import org.apache.commons.lang.StringUtils;
+import org.openmrs.Location;
+import org.openmrs.LocationAttribute;
+import org.openmrs.api.APIException;
+import org.openmrs.api.context.Context;
+
+public class LocationBasedPrefixProvider implements PrefixProvider {
```

Comment:
Rather than having the JavaDoc on the method bringing the most of its logic, put a header to this class an explain what it does.